pemutil/pem.go

```diff
@@ -277,6 +278,56 @@ func ParseCertificateRequest(pemData []byte) (*x509.CertificateRequest, error) {
 	return nil, errors.New("error parsing certificate request: no certificate found")
 }
 
+// PEMType represents a PEM block type. (e.g., CERTIFICATE, CERTIFICATE REQUEST, etc.)
+type PEMType int
+
+func (pt PEMType) String() string {
+	switch pt {
+	case PEMTypeCertificate:
+		return "certificate"
+	case PEMTypeCertificateRequest:
+		return "certificate request"
+	default:
+		return "undefined"
+	}
+}
+
+const (
+	// PEMTypeUndefined undefined
+	PEMTypeUndefined = iota
+	// PEMTypeCertificate CERTIFICATE
+	PEMTypeCertificate
+	// PEMTypeCertificateRequest CERTIFICATE REQUEST
+	PEMTypeCertificateRequest
+)
+
+// InvalidPEMError represents an error that occurs when parsing a file with
+// PEM encoded data.
+type InvalidPEMError struct {
+	Type    PEMType
+	File    string
+	Message string
+	Err     error
+}
+
+func (e *InvalidPEMError) Error() string {
+	switch {
+	case e.Message != "":
+		return e.Message
+	case e.Err != nil:
+		return e.Err.Error()
+	default:
+		if e.Type == PEMTypeUndefined {
+			return fmt.Sprintf("file %s does not contain valid PEM encoded data", e.File)
+		}
+		return fmt.Sprintf("file %s does not contain a valid PEM encoded %s", e.File, e.Type)
```

Comment:
It works for now, but the file might not be present if we add this error to methods like `ParseCertificate` where the arguments are just the PEM bytes.